src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2669,3 +2858,739 @@ def costing_upper_bound_eq(c, p):
 
         # method has finished building components
         b.components_already_built = True
+
+    def calculate_NPV(b, fixed_OM, variable_OM):
+        """
+        Equations for cash flow expressions derived from the textbook
+        Engineering Economy: Applying Theory to Practice, 3rd Ed. by Ted. G. Eschenbach.
+
+        The net present value (NPV) is a representative measure of the "current
+        day" value of a chemical plant over the total lifetime, including all
+        cash flows. The basic NPV formula is
+
+        NPV = Cash Flow In - Cash Flow Out
+
+        where Cash Flow In is revenue from product sales or salvage, and
+        Cash Flow Out is costs from capital, operating, royalties and loans.
+
+        This method supports capital expenditure, inflation, royalties,
+        loan repayment and capital depreciation. If these additional details
+        are not included, the NPV formulation assumes that negative cash flows
+        consists only of capital and operating costs that remain numerically
+        constant over time, and the resulting NPV formula is
+
+        NPV = [(REVENUE - OPEX) * P/A(r, N)] - CAPEX - Other_costs
+
+        where P/A(r, N) is the series present worth factor; this factor scales
+        a future cost to its present value from a known discount rate r and
+        project lifetime N based on annuity growth over time. This factor is
+        calculated as
+
+        P/A(r, N) = [ 1 - (1+r)**(-N) ] / r
+
+        where r is expressed as a decimal and N is expressed in years. In the
+        NPV expression above, REVENUE is the constant annual revenue, OPEX is
+        the constant annual operating cost, and CAPEX is the total capital cost.
+        Other_costs includes royalties, loan repayment, and capital depreciation
+        losses that are described in further detail below.
+
+        In the expressions above, revenue and operating costs are assumed to
+        remain numerically constant over time. Operating costs and revenues are
+        then adjusted based on predicted annuity growth to obtain the present
+        value. These expressions are implemented if there is no capital
+        expenditure period.
+
+        ----------------------------------------------------------------------
+
+        The general NPV formulation allows capital costs, operating costs,
+        and revenues to escalate over time via geometric gradient growth, e.g.
+        a constant proportional growth rate expressed as an escalation or
+        inflation percentage. Additionally, the formulation includes negative
+        cash flows from royalties, loan repayment and capital depreciation.
+        The general formulation is given by
+
+        NPV = PV_Revenue - PV_Operating_Cost - PV_Royalties - PV_Capital_Cost
+              - Loan_Interest_Owed - Capital_Depreciation
+
+        For escalating costs, the series present worth factor is modified to
```

Comment:
I would change this to say "for costs escalating at a constant rate for the project lifetime..." to help distinguish it from capital escalation